tensorboard/components/vz_bar_chart/vz-bar-chart.ts

```diff
@@ -0,0 +1,310 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+namespace vz_bar_chart {
+
+/**
+ * Encapsulates information for a single bar to rendered onto the chart.
+ */
+export interface Bar {
+  // The label on the X-axis for this bar.
+  x: string,
+
+  // The height of the bar.
+  y: number,
+}
+
+Polymer({
+  is: 'vz-bar-chart',
+  properties: {
+    /**
+     * How to feed data to the bar chart.
+     *
+     * Each key within the `data` object corresponds to a data series,
+     * each of which is associated with its own color of bars.
+     *
+     * Each entry within a list corresponds to an X-axis label (the string
+     * 'x' property) and bar height (The numeric 'y' property).
+     *
+     * Example:
+     * data = {'series0':[{ x: 'a', y: 1 }, { x: 'c', y: 3 }, { x: 'b', y: 2 }],
+     *        'series1':[{ x: 'a', y: 4 }, { x: 'g', y: 3 }, { x: 'e', y: 5 }]}
+     *
+     * This will generate a Plottable ClusteredBar chart with two series and
+     * 5 distinct classes ('a', 'b', 'c', 'g', 'e').
+     */
+    data: Object,
+
+    /**
+     * Scale that maps series names to colors. The default colors are from
+     * d3.schemeCategory10. Use this property to replace the default coloration.
+     *
+     * Note that if a `colorScale` gets passed in, it gets mutated within this
+     * component to have its domain set to the sorted keys of the `data` object.
+     * e.g. .domain(['series0', 'series1'])
+     * @type {Plottable.Scales.Color}
+     */
+    colorScale: {
+      type: Object,
+      value: function() {
+        return new Plottable.Scales.Color().range(d3.schemeCategory10);
+      }
+    },
+
+    /**
+     * How to format the tooltip.
+     *
+     * There should be a formatting object for each desired column.
+     * Each formatting object gets applied to the datum bound to the closest
+     * clustered bar.
+     */
+    tooltipColumns: {
+      type: Array,
+      value: function() {
+        return [
+          {
+            title: 'Name',
+            evaluate: function(d) {
+              return d.key;
+            },
+          },
+          {
+            title: 'X',
+            evaluate: function(d) {
+              return d.value.x;
+            },
+          },
+          {
+            title: 'Y',
+            evaluate: function(d) {
+              return d.value.y;
+            },
+          },
+        ];
+      }
+    },
+    _attached: Boolean,
+    _chart: Object,
+
+    /**
+     * Tooltip header innerHTML text. We cannot use a dom-repeat inside of a
```

Comment:
Is there any particular reason we compute the table headers as raw HTML and set inner-h-t-m-l for those, but the rest of the tooltip chart we build up dynamically with D3 selections?

Reply:
Indeed, lets use d3 in both places.